contracts/ERC7432/interfaces/IERC7432.sol

```diff
@@ -7,48 +7,82 @@ import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol
 
 /// @title ERC-7432 Non-Fungible Token Roles
 /// @dev See https://eips.ethereum.org/EIPS/eip-7432
-/// Note: the ERC-165 identifier for this interface is 0xeec2fffb.
+/// Note: the ERC-165 identifier for this interface is 0xd7e151ef.
 interface IERC7432 is IERC165 {
     struct RoleData {
         uint64 expirationDate;
         bytes data;
     }
 
+    /** Events **/
+
     /// @notice Emitted when a role is granted.
     /// @param _role The role identifier.
     /// @param _tokenAddress The token address.
     /// @param _tokenId The token identifier.
-    /// @param _grantee The user that receives the role assignment.
-    /// @param _expirationDate The expiration date of the role assignment.
-    /// @param _data Any additional data about the role assignment.
+    /// @param _grantor The user assigning the role.
+    /// @param _grantee The user receiving the role.
+    /// @param _expirationDate The expiration date of the role.
+    /// @param _data Any additional data about the role.
     event RoleGranted(
         bytes32 indexed _role,
         address indexed _tokenAddress,
         uint256 indexed _tokenId,
+        address _grantor,
         address _grantee,
-        uint64  _expirationDate,
+        uint64 _expirationDate,
         bytes _data
     );
 
     /// @notice Emitted when a role is revoked.
     /// @param _role The role identifier.
     /// @param _tokenAddress The token address.
     /// @param _tokenId The token identifier.
+    /// @param _revoker The user revoking the role.
     /// @param _grantee The user that receives the role revocation.
     event RoleRevoked(
         bytes32 indexed _role,
         address indexed _tokenAddress,
         uint256 indexed _tokenId,
+        address _revoker,
         address _grantee
     );
 
+    /// @notice Emitted when a user is approved to manage any role on behalf of another user.
+    /// @param _tokenAddress The token address.
+    /// @param _approver The user approving the operator.
+    /// @param _operator The user approved to grant and revoke roles.
+    /// @param _isApproved The approval status.
+    event RoleApprovalForAll(
+        address indexed _tokenAddress,
+        address indexed _approver,
+        address indexed _operator,
+        bool _isApproved
+    );
+
+    /// @notice Emitted when a user is approved to manage the roles of an NFT on behalf of another user.
+    /// @param _tokenAddress The token address.
+    /// @param _tokenId The token identifier.
+    /// @param _approver The user approving the operator.
+    /// @param _operator The user approved to grant and revoke roles.
+    /// @param _isApproved The approval status.
+    event RoleApproval(
+        address indexed _tokenAddress,
+        uint256 indexed _tokenId,
+        address indexed _approver,
+        address _operator,
+        bool _isApproved
```

Comment:
Do we need the approver on these events? It will always be the `msg.sender` on this case